File: app/src/main/java/com/task/data/remote/ServiceResponse.java
package com.task.data.remote;

/**
 * Created by AhmedEltaher on 5/12/2016
 */

public class ServiceResponse {
    private int code;
    private Object data;
    private ServiceError ServiceError;

    public ServiceResponse(int code, Object response) {
        this.code = code;
        this.data = response;
    }

    public ServiceResponse(ServiceError ServiceError) {
        this.ServiceError = ServiceError;
    }

    public ServiceResponse(Object response) {
        this.data = response;
    }

    public int getCode() {
        return code;
    }

    public ServiceError getServiceError() {
        return ServiceError;
    }

    public Object getData() {

        return data;
    }
}

File: app/src/main/java/com/task/data/remote/ServiceResponse.kt
package com.task.data.remote

/**
 * Created by AhmedEltaher on 5/12/2016
 */

class ServiceResponse(var code: Int = -1, var serviceError: ServiceError? = null, var data: Any? = null) {

    constructor(errorCode: Int, response: Any?) : this(errorCode, data = response)

    constructor(error: ServiceError) : this()

    constructor(response: Any) : this(data = response)
}
